### PR TITLE
Batch exchange rename

### LINKED ISSUE #333: [Easy-ish] Rename the StablecoinConverter
Although stable coins are the target objective for our PoC release, this contract is not strictly bound to facilitate the exchange of only stable coins. That is to say, that the current contract naming may be confusing or misleading to some readers and should be changed. Among the following options discussed, 

- `BatchExchange`
- `TokenConverter`
- `ExchangeEngine`
- `BatchEngine`

the [dƒusion team](url) has decided to change the name of `StablecoinConverter` to `BatchExchange`.

Note that such a simple task will affect two other related repositories. Namely 
1. [dex-react](https://github.com/gnosis/dex-react) - @anxo, @W3stside, @alfetopito, @Velenir 
2. [dex-services](https://github.com/gnosis/dex-services) - @josojo, @fleupold, @nlordell 

Which both depend on this contract!

### PR DESCRIPTION
**Pull request by josojo**
closes #333 

---
testplan:

check out that all the dependent repos also work
https://github.com/gnosis/dex-services/pull/314
https://github.com/gnosis/dex-react/pull/263

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
All green ==> All good!

File: networks.json [hunk 1]
```diff
@@ -404,7 +404,7 @@
       }
     }
   },
-  "StablecoinConverter": {
+  "BatchExchange": {
```
Comment (fleupold) [inline networks.json:407]: Do we need to redeploy?

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F4PNIwPRJufPDW%2Fgiphy.gif&token=find_replace" data-gitmeme-token="find_replace"></a>

File: networks.json (hunk 1 above)
Comment (josojo) [reply networks.json:407]: nope, not for that...

It is okay to rename it and keep all contract data the same. 
Just republishing the npm package might be needed

**Pull request merged**